### PR TITLE
`azurerm_stream_analytics_job` Removing requirement of optional properties

### PR DESCRIPTION
**Pull request by nexxai**
Fixes https://github.com/terraform-providers/terraform-provider-azurerm/issues/3342

### CODE REVIEW

File: azurerm/resource_arm_stream_analytics_job.go [hunk 1]
```diff
@@ -42,7 +42,7 @@ func resourceArmStreamAnalyticsJob() *schema.Resource {
 
 			"compatibility_level": {
 				Type:     schema.TypeString,
-				Required: true,
+				Optional: true,
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_stream_analytics_job.go]: 🤔 on reflection could we update the `TestAccAzureRMStreamAnalyticsJob_basic` test to skip these properties (and add a `TestAccAzureRMStreamAnalyticsJob_complete` test with these set - to ensure that we're verifying both use-cases work?

File: azurerm/resource_arm_stream_analytics_job.go [hunk 2]
```diff
@@ -137,7 +137,7 @@ func resourceArmStreamAnalyticsJobCreateUpdate(d *schema.ResourceData, meta inte
 	}
 
 	compatibilityLevel := d.Get("compatibility_level").(string)
-	dataLocale := d.Get("data_locale").(string)
+	// dataLocale := d.Get("data_locale").(string)
```
Comment (katbyte) [inline azurerm/resource_arm_stream_analytics_job.go]: This is going to break the property?

File: azurerm/resource_arm_stream_analytics_job.go [hunk 3]
```diff
@@ -163,8 +163,8 @@ func resourceArmStreamAnalyticsJobCreateUpdate(d *schema.ResourceData, meta inte
 			Sku: &streamanalytics.Sku{
 				Name: streamanalytics.Standard,
 			},
-			CompatibilityLevel:                 streamanalytics.CompatibilityLevel(compatibilityLevel),
-			DataLocale:                         utils.String(dataLocale),
+			CompatibilityLevel: streamanalytics.CompatibilityLevel(compatibilityLevel),
+			// DataLocale:                         utils.String(dataLocale),
```
Comment (katbyte) [inline azurerm/resource_arm_stream_analytics_job.go]: And here

File: azurerm/resource_arm_stream_analytics_job.go (hunk 1 above)
Comment (katbyte) [reply azurerm/resource_arm_stream_analytics_job.go]: Agree with this sentiment

**Comment by nexxai** [comment]
I had just made those changes temporarily while @mbfrahry and I were trying to troubleshoot the problem.  He and I have a pair programming session set up for Wed. afternoon to figure out the issue.

**Comment by nexxai** [comment]
After session with @mbfrahry all tests are passing.  Documentation is updated.

File: azurerm/resource_arm_stream_analytics_job.go (hunk 2 above)
Comment (nexxai) [reply azurerm/resource_arm_stream_analytics_job.go]: Fixed.

File: azurerm/resource_arm_stream_analytics_job.go (hunk 3 above)
Comment (nexxai) [reply azurerm/resource_arm_stream_analytics_job.go]: Fixed.

File: azurerm/resource_arm_stream_analytics_job.go (hunk 1 above)
Comment (nexxai) [reply azurerm/resource_arm_stream_analytics_job.go]: Done

**Pull request merged**